bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,687 @@
+            return mirekSchemaFrom(new Resource(resourceType).setId(resourceId));
+        }
+        return new MirekSchema();
+    }
+
+    /**
+     * Update the channel state depending on a new resource sent from the bridge.
+     *
+     * @param resource a Resource object containing the new state.
+     */
+    public void onResource(Resource resource) {
+        if (!disposing) {
+            logger.debug("onResource(..) {}", resource);
+            if (thisResource.getId().equals(resource.getId())) {
+                if (resource.hasFullState()) {
+                    thisResource = resource;
+                    if (!updatePropertiesDone) {
+                        updateProperties(resource);
+                    }
+                }
+                if (!updateDependenciesDone) {
+                    scheduler.submit(() -> updateDependencies());
+                }
+            }
+            String cacheId = resource.getId();
+            Resource cacheResource = contributorsCache.get(cacheId);
+            if (Objects.nonNull(cacheResource)) {
+                resource.copyMissingFieldsFrom(cacheResource);
+                updateChannels(resource);
+                contributorsCache.put(cacheId, resource);
+            }
+        }
+    }
+
+    /**
+     * If this v2 thing has a matching v1 legacy thing in the system, then for each channel in the v1 thing that
+     * corresponds to an equivalent channel in this v2 thing, and for all items that are linked to the v1 channel,
+     * create a new channel/item link between that item and the respective v2 channel in this thing.
+     */
+    private void updateChannelItemLinksFromLegacy() {
+        if (!disposing) {
+            synchronized (legacyLinkedChannelUIDs) {
+                legacyLinkedChannelUIDs.forEach(legacyLinkedChannelUID -> {
+                    String targetChannelId = HueBindingConstants.REPLICATE_CHANNEL_ID_MAP
+                            .get(legacyLinkedChannelUID.getId());
+                    if (Objects.nonNull(targetChannelId)) {
+                        Channel targetChannel = thing.getChannel(targetChannelId);
+                        if (Objects.nonNull(targetChannel)) {
+                            ChannelUID uid = targetChannel.getUID();
+                            itemChannelLinkRegistry.getLinkedItems(legacyLinkedChannelUID).forEach(linkedItem -> {
+                                String item = linkedItem.getName();
+                                if (!itemChannelLinkRegistry.isLinked(item, uid)) {
+                                    logger.info("Created link between Channel:{} and Item:{}", uid, item);
+                                    itemChannelLinkRegistry.add(new ItemChannelLink(item, uid));
+                                }
+                            });
+                        }
+                    }
+                });
+                legacyLinkedChannelUIDs.clear();
+            }
+        }
+    }
```

Comment:
@jlaur it works as follows..

1. If you have an existing API v1 installation, with Things, Channels, Items, then =>
2. If the discovering process finds an API v2 Device thing, then =>
3. If the discovered v2 Device thing has an IDV1 tag, then =>
4. If the IDV1 tag matches an existing API V1 Light thing, then =>
5. Add the existing API V1 Light thing UID as a property to the discovered v2 Device thing in the Inbox..
..
7. If the discovered v2 Device thing in the Inbox is instantiated, then =>
8. Check if it has a property for an existing API V1 Light thing UID, then =>
9. Parse the instantiated v2 Device thing's channels, and the existing API V1 Light thing's channels, and =>
10. If the both the new thing and the existing thing have the same channels, then =>
11. Parse the items that are linked to the existing thing's channels, and =>
12. Link those same items to the new thing's corresponding channel

So to answer your questions

> Will it create links between channels and items that doesn't actually exist? 

No.

> Will this always be desired

I think so :) -- and obviously if the link created in step 12. above is not desired, then it can always be un-linked again..

> and how will it work for file-based configuration?

It works for existing items and things that are either UI created, or file created. But due to the discovery process involvement, the above process is only followed if the new thing is instantiated from the Inbox.

> And how will it work for managed things, will this be persisted in jsondb?

If everything is UI created, then everything is persisted. If the new Thing is file based, then the process does not happen. And it the existing Items are file based, then the links are not persisted.